staging/src/k8s.io/apiextensions-apiserver/pkg/controller/openapi/builder/merge.go

```diff
@@ -83,28 +88,29 @@ func mergeSpec(dest, source *spec.Swagger) {
 }
 
 // MergeSpecsV3 merges OpenAPI v3 specs for CRDs
-// For V3, the static spec is never merged with the individual CRD specs so no conflict resolution is necessary
+// Conflicts belonging to the meta.v1 or autoscaling.v1 group versions are skipped as all CRDs reference those types
+// Other conflicts will result in an error
 func MergeSpecsV3(crdSpecs ...*spec3.OpenAPI) (*spec3.OpenAPI, error) {
-	// create shallow copy of staticSpec, but replace paths and definitions because we modify them.
 	crdSpec := &spec3.OpenAPI{}
 	if len(crdSpecs) > 0 {
 		crdSpec.Version = crdSpecs[0].Version
 		crdSpec.Info = crdSpecs[0].Info
 	}
 	for _, s := range crdSpecs {
-		// merge specs without checking conflicts, since the naming controller prevents
-		// conflicts between user-defined CRDs
-		mergeSpecV3(crdSpec, s)
+		err := mergeSpecV3(crdSpec, s)
+		if err != nil {
```

Comment:
erroring out rather than warn seems great, as we would catch these things earlier.